### PR TITLE
feat: add new built_with_electron variable to config.gypi

### PR DESCRIPTION
**Pull request by codebytere**
Resolves https://github.com/electron/electron/issues/6556.

Adds new `built_with_electron` variable to `config.gypi` generated by `tools/generate_config_gypi.py`

/cc @nornagon

### CODE REVIEW

**Comment by refack** [comment]
IMHO it fits better in `config.gypi` which is also exposed to `node-gyp` (and is also exposed via `process.config`).

File: tools/generate_config_gypi.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 def main(args):
   out = args[0]
   with open(out, 'w') as f:
-    f.write("{'variables':{}}\n")
+    f.write("{'variables':['built_with_electron': 1]}\n")
```
Comment (nornagon) [inline tools/generate_config_gypi.py]: errr, should this be `{}`? this doesn't look like it'll parse

Comment (refack) [reply tools/generate_config_gypi.py]: Yep, should be `f.write("{'variables':{'built_with_electron': 1}}\n")`

Comment (ryzokuken) [reply tools/generate_config_gypi.py]: @codebytere did this on my suggestion which was in part based on https://gyp.gsrc.io/docs/InputFormatReference.md#Variables-and-Conditionals

**Pull request merged**